.github/workflows/parallel.yml

```diff
@@ -105,3 +109,10 @@ jobs:
       continue-on-error: true
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
+    - name: Upload test logs
+      if: failure()
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ inputs.tests-name }}-log-${{ matrix.profile }}
+        path: |
```

Comment:
> Nit: we may also want to set the `retention-days`, such as 7. Otherwise 90 days by default is too long.

How about setting it for all upload actions in new PR? I think we also need to upgrade some action versions.

![image](https://github.com/apache/incubator-uniffle/assets/17894939/823dba18-1f93-4138-bc5c-0f805287cea3)